src/backend/api/Fusion.Resources.Logic/Requests/Commands/ResourceAllocationRequest/Allocation/CanApproveStepHandler.cs

```diff
@@ -1,42 +1,79 @@
-﻿using Fusion.Resources.Database.Entities;
+﻿using Fusion.Resources.Database;
+using Fusion.Resources.Database.Entities;
+using Fusion.Resources.Logic.Requests;
+using Fusion.Resources.Logic.Workflows;
 using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore;
+using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
-using Fusion.Resources.Database;
 
 namespace Fusion.Resources.Logic.Commands
 {
     public partial class ResourceAllocationRequest
     {
-        public class CanApproveStepHandler : INotificationHandler<CanApproveStep>
+        public class CanApproveStepHandler : CanApproveStepHandlerBase, INotificationHandler<CanApproveStep>
         {
+            private static readonly Dictionary<WorkflowAccessKey, WorkflowAccess> AccessTable = new Dictionary<WorkflowAccessKey, WorkflowAccess>
```

Comment:
Any chance we could move this into the workflow definition? I guess we can add that to a refactor task.. Maybe align with the fluent structure used there.

Reply:
I'll refactor this in another branch.